crates/sqlbuiltins/src/functions/scalars/unwind.rs

```diff
@@ -0,0 +1,180 @@
+use bson::Document;
+use datafusion::arrow::datatypes::Fields;
+use serde_json::{from_str, Map, Value};
+
+use super::*;
+use datasources::bson::scalar as bson_scalar;
+
+pub struct Json;
+
+impl ConstBuiltinFunction for Json {
+    const NAME: &'static str = "unwind_json";
+    const DESCRIPTION: &'static str =
+        "Converts a json document in a string field to a struct field.";
+    const EXAMPLE: &'static str = "unwind_json(<value>)";
+    const FUNCTION_TYPE: FunctionType = FunctionType::Scalar;
+
+    fn signature(&self) -> Option<Signature> {
+        Some(Signature::new(
+            // args: <FIELD>
+            TypeSignature::OneOf(vec![
+                TypeSignature::Exact(vec![DataType::Utf8]),
+                TypeSignature::Exact(vec![DataType::LargeUtf8]),
+            ]),
+            Volatility::Immutable,
+        ))
+    }
+}
+
+impl BuiltinScalarUDF for Json {
+    fn as_expr(&self, args: Vec<Expr>) -> Expr {
+        let udf = ScalarUDF {
+            name: Self::NAME.to_string(),
+            signature: ConstBuiltinFunction::signature(self).unwrap(),
+            return_type: Arc::new(|_| Ok(Arc::new(DataType::Struct(Fields::empty())))),
```

Comment:
I'm pretty sure this'll break during logical planning due to datafusion's logical planner not having the correct return type.

datafusion doesn't yet support dynamic return types for udfs. 

See https://github.com/apache/arrow-datafusion/issues/8624

This will likely have to be done outside of the 
`BuiltinScalarUDF` similar to how [arrow_cast](https://github.com/apache/arrow-datafusion/blob/b85a39739e754576723ff4b1691c518a86335769/datafusion/sql/src/expr/arrow_cast.rs#L55-L56) is handled.

There is a chance that datafusion planner doesn't check struct fields during logical planning in which case we are ok, but we'll want to extensively test this.

Reply:
noted.